src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -621,10 +621,19 @@ def get_user_member_groups(cmd, upn_or_object_id, security_enabled_only=False):
 
 def create_group(cmd, display_name, mail_nickname):
     graph_client = _graph_client_factory(cmd.cli_ctx)
+    matches = list(graph_client.groups.list(filter="displayName eq '{}' and mailNickname eq '{}'".format(
+        display_name, mail_nickname)))
+
+    # workaround to ensure idempotent even AAD graph service doesn't support it
+    if matches:
+        if len(matches) > 1:
+            err = ('There are more than one groups with same display name and mail nick names: "{}". '
+                   'Please delete them first.')
+            raise CLIError(err.format(', '.join([x.object_id for x in matches])))
```

Comment:
Given that the service doesn't create this restriction, and in the past we've allowed Groups with the same display/nick names, I think out-right preventing people from doing it is a step too far. I'd feel better if there was a flag to allow a "force" that disregards this check.